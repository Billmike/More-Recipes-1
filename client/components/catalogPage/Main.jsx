import React from 'react';
import PropTypes from 'prop-types';
import queryString from 'query-string';

import SearchBar from '../common/SearchBar';
import SearchResult from './SearchResult';
import CatalogResult from './CatalogResult';

const propTypes = {
  recipes: PropTypes.shape({
    searchResult: PropTypes.arrayOf(PropTypes.shape()),
  }),
  location: PropTypes.shape({
    search: PropTypes.string,
  }),
  searchedTerm: PropTypes.string,
  showText: PropTypes.bool.isRequired,
};

const defaultProps = {
  recipes: undefined,
  searchedTerm: undefined,
  location: undefined,
};

/**
 * Function representing main component
 *
 * @param {object} props - The properties passed to the component
 *
 * @returns {object} React element
 */
const Main = (props) => {
  const { recipes: { searchResult }, location: { search }, showText } = props;
  let { searchedTerm } = props;

  if (search !== '') {
    const parsed = queryString.parse(search);
    searchedTerm = parsed.list;
  }
  return (
    <div>
      <SearchBar {...props} />
      <div className="row recipes-list">
        <div className="center-align">
          {showText &&
            <h5><em>{searchResult.length === 0
              ? `No Search result found for ${searchedTerm}`
              : `Search results for ${searchedTerm}`
            }</em>
            </h5>
          }
        </div>
        <div>
          {showText
            ? <SearchResult {...props} />
            : <CatalogResult {...props} />}
        </div>

      </div>
    </div>
  );
};

Main.propTypes = propTypes;
Main.defaultProps = defaultProps;

export default Main;
