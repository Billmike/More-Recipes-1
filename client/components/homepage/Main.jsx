import React from 'react';
import PropTypes from 'prop-types';

import PopularRecipesCatalog from './PopularRecipesCatalog';
import AllRecipeCatalog from './AllRecipeCatalog';
import SearchBar from '../common/SearchBar';
import Slider from './Slider';

const propTypes = {
  recipes: PropTypes.shape({
    recipes: PropTypes.arrayOf(PropTypes.shape()),
    popularRecipes: PropTypes.arrayOf(PropTypes.shape()),
  }),
};

const defaultProps = {
  recipes: undefined,
};

/**
 * Main component
 *
 * @param {object} props - The properties passed to the component
 *
 * @returns {object} React element
 */
const Main = (props) => {
  const { recipes: { recipes, popularRecipes } } = props;

  return (
    <div>
      <div id="position">
        <Slider />
        <SearchBar
          id={'home-search-bar'}
          {...props}
        />
      </div>
      <div className="grey lighten-3" >
        <div className="row recipes-list">
          <div className="col s12 popular-list white">
            <PopularRecipesCatalog popularRecipes={popularRecipes} {...props} />
          </div>
          <div className="col s12 all-recipes-list white">
            <AllRecipeCatalog recipesCatalog={recipes} />
          </div>
        </div>
      </div>
    </div>
  );
};

Main.propTypes = propTypes;
Main.defaultProps = defaultProps;

export default Main;
