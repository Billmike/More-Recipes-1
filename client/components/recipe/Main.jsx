import React from 'react';

import RecipeImage from '../common/RecipeImage';
import RecipeCard from './RecipeCard';
import RecipeDetails from './RecipeDetails';
import ReviewCollection from './ReviewCollection';
import AddReview from './AddReview';

const Main = () => (
  <main>
    <div className="parallax-container">
      <div className="parallax">
        <RecipeImage />
      </div>
    </div>
    <div className="section white container">
      <div className="row">
        <div className="col s12">
          <h5 className="header">German Sauce</h5>
        </div>
      </div>
      <div className="row">
        <div className="col s12 m6 l5">
          <RecipeCard />
        </div>
        <div className="col s12 m6 l7">
          <RecipeDetails />
        </div>
      </div>
      <ReviewCollection />
      <AddReview />
    </div>
  </main>
);

export default Main;
