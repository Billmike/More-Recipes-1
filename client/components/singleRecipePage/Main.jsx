import React from 'react';
import PropTypes from 'prop-types';

import RecipeImage from '../common/RecipeImage';
import RecipeCard from './RecipeCard';
import RecipeDetails from './RecipeDetails';
import ReviewCollection from './ReviewCollection';
import AddReview from './AddReview';

const propTypes = {
  singleRecipe: PropTypes.shape({
    recipe: PropTypes.shape(),
    reviews: PropTypes.arrayOf(PropTypes.shape()),
    favoritedUsers: PropTypes.arrayOf(PropTypes.shape()),
    voters: PropTypes.arrayOf(PropTypes.shape()),
    voteMessage: PropTypes.string,
    favoriteMessage: PropTypes.string,
  }).isRequired,
};

const Main = (props) => {
  const { singleRecipe } = props;
  const { recipe, reviews, favoritedUsers, voters, voteMessage,
    favoriteMessage } = singleRecipe;
  const { title, recipeImage } = recipe;

  return (
    <div>
      <div className="parallax-container">
        <div className="parallax">
          <RecipeImage title={title} recipeImage={recipeImage} />
        </div>
      </div>
      <div className="section white container">
        <div className="row">
          <div className="col s12">
            <h4 className="title-header">{title}</h4>
          </div>
        </div>
        <div className="row">
          <div className="col s12 m6 l5">
            <RecipeCard
              recipe={recipe}
              favoritedUsers={favoritedUsers}
              voters={voters}
              voteMessage={voteMessage}
              favoriteMessage={favoriteMessage}
              {...props}
            />
          </div>
          <div className="col s12 m6 l6 offset-l1">
            <RecipeDetails recipe={recipe} />
          </div>
        </div>
        <ReviewCollection reviews={reviews} />
        <AddReview {...props} />
      </div>
    </div>
  );
};

Main.propTypes = propTypes;

export default Main;
