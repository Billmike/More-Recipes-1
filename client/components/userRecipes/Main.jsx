import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import UserAddedRecipes from './UserAddedRecipes';
import UserFavoriteCard from './UserFavoriteCard';
import EditRecipe from './EditRecipe';
import DeleteRecipe from './DeleteRecipe';
import AddRecipe from './AddRecipe';

/**
 * Class representing the main component
 *
 * @class Main
 * @extends {React.Component}
 */
class Main extends React.Component {
  /**
   * Creates an instance of UserRecipes.
   *
   * @memberof Main
   */
  constructor() {
    super();
    this.state = {
      openEdit: false,
      openDelete: false,
      openAdd: false,
      imagePreview: '',
    };
    this.handleOpenEdit = this.handleOpenEdit.bind(this);
    this.handleOpenDelete = this.handleOpenDelete.bind(this);
    this.handleOpenAdd = this.handleOpenAdd.bind(this);
    this.handleClose = this.handleClose.bind(this);
    this.handleImagePreview = this.handleImagePreview.bind(this);
  }

  /**
   * Opens the edit recipe modal
   *
   * @returns {object} Set open state to true
   * @memberof Main
   */
  handleOpenEdit() {
    this.setState({ openEdit: true });
  }

  /**
   * Opens the delete recipe modal
   *
   * @returns {object} Set openDelete state to true
   * @memberof Main
   */
  handleOpenDelete() {
    this.setState({ openDelete: true });
  }

  /**
   * Opens the add recipe modal
   *
   * @returns {object} Set openAdd state to true
   * @memberof Main
   */
  handleOpenAdd() {
    this.setState({ openAdd: true });
  }

  /**
   * Closes the modal
   *
   * @returns {object} Set open state to false
   * @memberof Main
   */
  handleClose() {
    this.setState({
      openEdit: false,
      openDelete: false,
      openAdd: false,
      imagePreview: '',
    });
  }

  /**
   * Funtion to handle image preview
   *
   * @param {any} imagePreview - The image to be displaced
   * @returns {objec} Set the state property imagePreview
   * @memberof Main
   */
  handleImagePreview(imagePreview) {
    console.log(imagePreview);
    this.setState({
      imagePreview,
    });
  }

  /**
   * Render method
   *
   * @returns {object} React element
   * @memberof Main
   */
  render() {
    return (
      <div className="row">
        <div className="fixed-action-btn">
          <a
            role="button"
            tabIndex="0"
            onClick={this.handleOpenAdd}
            className={`btn-floating btn-large modal-trigger waves-effect
              waves-light indigo accent-2 pulse`}
          >
            <i className="material-icons">add</i>
          </a>
        </div>
        <div className="my-recipes">
          <div id="user-recipes">
            <UserAddedRecipes
              handleOpenEdit={this.handleOpenEdit}
              handleOpenDelete={this.handleOpenDelete}
              {...this.props}
            />
          </div>
          <div id="user-favorites" className="col s12">
            <UserFavoriteCard />
          </div>
        </div>
        <MuiThemeProvider>
          <EditRecipe
            open={this.state.openEdit}
            handleClose={this.handleClose}
          />
        </MuiThemeProvider>
        <MuiThemeProvider>
          <DeleteRecipe
            open={this.state.openDelete}
            handleClose={this.handleClose}
          />
        </MuiThemeProvider>
        <MuiThemeProvider>
          <AddRecipe
            open={this.state.openAdd}
            handleClose={this.handleClose}
            imagePreview={this.state.imagePreview}
            handleImagePreview={this.handleImagePreview}
            {...this.props}
          />
        </MuiThemeProvider>
      </div>
    );
  }
}

export default Main;
