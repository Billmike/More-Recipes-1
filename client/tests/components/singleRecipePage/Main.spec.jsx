import React from 'react';

import Main from '../../../components/singleRecipePage/Main';

describe('Main component', () => {
  let spy;

  afterEach(() => {
    spy.mockClear();
  });
  it('should render component', () => {
    spy = jest.spyOn(Main.prototype, 'componentDidMount');
    const props = {
      singleRecipe: {
        recipe: {},
        reviews: [],
        favoritedUsers: [],
        voters: [],
        voteMessage: 'Upvote recorded',
        favoriteMessage: 'Recipe added to favorites',
        hasMoreReviews: false,
        isLoadingReviews: false,
      },
      reviewContent: 'Review for a recipe',
      handleViewMoreReviews: jest.fn(),
      handleUpvote: jest.fn(),
      handleDownvote: jest.fn(),
      handleFavorite: jest.fn(),
    };
    const wrapper = shallow(<Main {...props} />);

    expect(wrapper).toBeDefined();
    expect(spy).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render component with spinner', () => {
    const props = {
      singleRecipe: {
        recipe: {},
        reviews: [],
        favoritedUsers: [],
        voters: [],
        voteMessage: 'Upvote recorded',
        favoriteMessage: 'Recipe added to favorites',
        hasMoreReviews: false,
        isLoadingReviews: true,
      },
      reviewContent: 'Review for a recipe',
      handleViewMoreReviews: jest.fn(),
      handleUpvote: jest.fn(),
      handleDownvote: jest.fn(),
      handleFavorite: jest.fn(),
    };
    const wrapper = shallow(<Main {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('Spinner').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
