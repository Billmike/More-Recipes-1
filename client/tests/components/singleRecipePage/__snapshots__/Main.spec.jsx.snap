// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should render component 1`] = `
<div>
  <div
    className="parallax-container"
  >
    <div
      className="parallax"
    >
      <RecipeImage
        recipeImage={null}
        title={null}
      />
    </div>
  </div>
  <div
    className="section white container"
  >
    <div
      className="row"
    >
      <div
        className="col s12"
      >
        <h4
          className="title-header"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col s12 m6 l5"
      >
        <RecipeCard
          favoriteMessage={null}
          handleDownvote={[MockFunction]}
          handleFavorite={[MockFunction]}
          handleUpvote={[MockFunction]}
          handleViewMoreReviews={[MockFunction]}
          reviewContent="Review for a recipe"
          singleRecipe={
            Object {
              "favoriteMessage": "Recipe added to favorites",
              "favoritedUsers": Array [],
              "hasMoreReviews": false,
              "isLoadingReviews": false,
              "recipe": Object {},
              "reviews": Array [],
              "voteMessage": "Upvote recorded",
              "voters": Array [],
            }
          }
          voteMessage={null}
        />
      </div>
      <div
        className="col s12 m6 l6 offset-l1"
      >
        <RecipeDetails
          recipe={Object {}}
        />
      </div>
    </div>
    <AddReview
      handleDownvote={[MockFunction]}
      handleFavorite={[MockFunction]}
      handleUpvote={[MockFunction]}
      handleViewMoreReviews={[MockFunction]}
      reviewContent="Review for a recipe"
      singleRecipe={
        Object {
          "favoriteMessage": "Recipe added to favorites",
          "favoritedUsers": Array [],
          "hasMoreReviews": false,
          "isLoadingReviews": false,
          "recipe": Object {},
          "reviews": Array [],
          "voteMessage": "Upvote recorded",
          "voters": Array [],
        }
      }
    />
    <ReviewCollection
      handleDownvote={[MockFunction]}
      handleFavorite={[MockFunction]}
      handleUpvote={[MockFunction]}
      handleViewMoreReviews={[MockFunction]}
      reviewContent="Review for a recipe"
      reviews={Array []}
      singleRecipe={
        Object {
          "favoriteMessage": "Recipe added to favorites",
          "favoritedUsers": Array [],
          "hasMoreReviews": false,
          "isLoadingReviews": false,
          "recipe": Object {},
          "reviews": Array [],
          "voteMessage": "Upvote recorded",
          "voters": Array [],
        }
      }
    />
    <div
      className="row center-align"
    >
      <div />
      <div>
        <button
          className="btn waves-effect waves-light indigo accent-2
                white-text"
          disabled={true}
          id="view-more-reviews"
          onClick={[MockFunction]}
        >
          View More
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Main component should render component with spinner 1`] = `
<div>
  <div
    className="parallax-container"
  >
    <div
      className="parallax"
    >
      <RecipeImage
        recipeImage={null}
        title={null}
      />
    </div>
  </div>
  <div
    className="section white container"
  >
    <div
      className="row"
    >
      <div
        className="col s12"
      >
        <h4
          className="title-header"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col s12 m6 l5"
      >
        <RecipeCard
          favoriteMessage={null}
          handleDownvote={[MockFunction]}
          handleFavorite={[MockFunction]}
          handleUpvote={[MockFunction]}
          handleViewMoreReviews={[MockFunction]}
          reviewContent="Review for a recipe"
          singleRecipe={
            Object {
              "favoriteMessage": "Recipe added to favorites",
              "favoritedUsers": Array [],
              "hasMoreReviews": false,
              "isLoadingReviews": true,
              "recipe": Object {},
              "reviews": Array [],
              "voteMessage": "Upvote recorded",
              "voters": Array [],
            }
          }
          voteMessage={null}
        />
      </div>
      <div
        className="col s12 m6 l6 offset-l1"
      >
        <RecipeDetails
          recipe={Object {}}
        />
      </div>
    </div>
    <AddReview
      handleDownvote={[MockFunction]}
      handleFavorite={[MockFunction]}
      handleUpvote={[MockFunction]}
      handleViewMoreReviews={[MockFunction]}
      reviewContent="Review for a recipe"
      singleRecipe={
        Object {
          "favoriteMessage": "Recipe added to favorites",
          "favoritedUsers": Array [],
          "hasMoreReviews": false,
          "isLoadingReviews": true,
          "recipe": Object {},
          "reviews": Array [],
          "voteMessage": "Upvote recorded",
          "voters": Array [],
        }
      }
    />
    <ReviewCollection
      handleDownvote={[MockFunction]}
      handleFavorite={[MockFunction]}
      handleUpvote={[MockFunction]}
      handleViewMoreReviews={[MockFunction]}
      reviewContent="Review for a recipe"
      reviews={Array []}
      singleRecipe={
        Object {
          "favoriteMessage": "Recipe added to favorites",
          "favoritedUsers": Array [],
          "hasMoreReviews": false,
          "isLoadingReviews": true,
          "recipe": Object {},
          "reviews": Array [],
          "voteMessage": "Upvote recorded",
          "voters": Array [],
        }
      }
    />
    <div
      className="row center-align"
    >
      <div>
        <Spinner
          size="small"
        />
      </div>
      <div>
        <button
          className="btn waves-effect waves-light indigo accent-2
                white-text"
          disabled={true}
          id="view-more-reviews"
          onClick={[MockFunction]}
        >
          View More
        </button>
      </div>
    </div>
  </div>
</div>
`;
